src/main/java/com/tomtom/services/configuration/implementation/Configuration.java

```diff
@@ -653,19 +647,107 @@ private static boolean checkNodeMatchStringsChildren(@Nullable final List<NodeDT
     }
 
     /**
+     * Check that all parameters have a key and a value.
+     * @param params list of parameters to check
+     * @throws IncorrectConfigurationException If a parameter was found without key or value.
+     */
+    private static void checkAllParamsHaveKeyAndValue(final List<ParameterDTO> params) throws IncorrectConfigurationException {
+        for (final ParameterDTO p : params) {
+            if ((p.getKey() == null) ||
+                (p.getValue() == null)) {
+                throw new IncorrectConfigurationException("Parameter found without key or value!");
+            }
+        }
+    }
+
+    /**
+     * Expand all the includes in a parameter.
+     *
+     * @param tree     Parameter to expand.
+     * @param included Memory of which include files were processed.
+     * @throws IncorrectConfigurationException If include recursion was detected.
+     * @return Replacements for the parameter that was just expanded.
```

Comment:
Run auto-format once in IDEA. This will place the parameters in the right order.